pkg/deployment/config.go

```diff
@@ -77,13 +81,18 @@ var (
 )
 
 func defaultConfig() *Config {
-	return &Config{
+	cfg := &Config{
 		ProgressDeadline:               ProgressDeadlineDefault,
 		DigestResolutionTimeout:        digestResolutionTimeoutDefault,
 		RegistriesSkippingTagResolving: sets.NewString("kind.local", "ko.local", "dev.local"),
 		QueueSidecarCPURequest:         &QueueSidecarCPURequestDefault,
 		ConcurrencyStateEndpoint:       "",
 	}
+	if cfg.QueueSidecarTokenAudiences == nil {
+		cfg.QueueSidecarTokenAudiences = sets.NewString("")
```

Comment:
I really don't think you need to do this -- assuming the set is initialized elsewhere, leaving it as `nil` should be fine for iteration. If you need to call `Insert` elsewhere, then `sets.NewString()` without any elements would be more correct (you happen to be filtering it out down below, but that seems like a bunch of additional work).

Example:

https://go.dev/play/p/nOyp5LLbOJ5

Reply:
We need this for ConfigMap testing,

The way the serving code works:
(1) You are expected to have an example in deployment.yaml -that is `queue-sidecar-token-audiences: ""`
(2) The example is used by testing.... and it is expected that the defaultConfig will be inline with the example!

So since we cant remove `queue-sidecar-token-audiences: ""` and since the defaultConfig without this extra code means cfg.QueueSidecarTokenAudiences == nil , tests fail without it.